pubtools/pulplib/_impl/model/repository/repo_lock.py

```diff
@@ -0,0 +1,163 @@
+import json
+import random
+import string
+from time import sleep
+import os
+import logging
+
+from datetime import datetime, timedelta
+from more_executors.futures import f_map
+
+LOCK_CLAIM_STR = "pulplib-lock-claim-"
+
+LOCK_VALID_FROM_OFFSET = os.getenv("PULPLIB_LOCK_VALID_FROM_OFFSET", 10)
```

Comment:
I don't think these actually work right now if you set the env vars.  There'll be a type error.

If you consider the expression:

```python
os.getenv("PULPLIB_LOCK_VALID_FROM_OFFSET", 10)
```

- if the caller didn't set the env var, it returns an int.
- if the caller set the env var, it returns a str (because env vars are always strings).

The code using these values unconditionally tries to use them as integers (e.g. using with 'timedelta' class).  So if you try to set one of these env vars and then run the example, I'm betting it'll crash.

Given that getenv will return a str in the 'env var is not set' case, it's best to make the expression *also* return a str in the 'env var is set' case, that way you can't forget to convert it to an int.

I usually prefer this pattern instead:

```suggestion
LOCK_VALID_FROM_OFFSET = int(os.getenv("PULPLIB_LOCK_VALID_FROM_OFFSET") or "10"))
```

...which will do the conversion properly in both set and unset cases; and also if the caller has set PULPLIB_LOCK_VALID_FROM_OFFSET to an empty string, this is handled as "use the default", rather than "crash because empty string can't be converted to int".

Reply:
Wow, it seems like pylint these days detects the same issue, which I didn't know about.  The `static` CI config currently complains about this:

```
pubtools/pulplib/_impl/model/repository/repo_lock.py:14:25: W1508: os.getenv default type is builtins.int. Expected str or None. (invalid-envvar-default)
```